crates/bevy_pbr/src/meshlet/visibility_buffer_raster_node.rs

```diff
@@ -379,35 +384,34 @@ fn downsample_depth(
     render_context: &mut RenderContext,
     meshlet_view_resources: &MeshletViewResources,
     meshlet_view_bind_groups: &MeshletViewBindGroups,
-    downsample_depth_pipeline: &RenderPipeline,
+    downsample_depth_first_pipeline: &ComputePipeline,
+    downsample_depth_second_pipeline: &ComputePipeline,
 ) {
-    render_context
-        .command_encoder()
-        .push_debug_group("meshlet_downsample_depth");
-
-    for i in 0..meshlet_view_resources.depth_pyramid_mips.len() {
-        let downsample_pass = RenderPassDescriptor {
-            label: Some("downsample_depth"),
-            color_attachments: &[Some(RenderPassColorAttachment {
-                view: &meshlet_view_resources.depth_pyramid_mips[i],
-                resolve_target: None,
-                ops: Operations {
-                    load: LoadOp::Clear(LinearRgba::BLACK.into()),
-                    store: StoreOp::Store,
-                },
-            })],
-            depth_stencil_attachment: None,
-            timestamp_writes: None,
-            occlusion_query_set: None,
-        };
+    let command_encoder = render_context.command_encoder();
+    let mut downsample_pass = command_encoder.begin_compute_pass(&ComputePassDescriptor {
+        label: Some("downsample_depth"),
+        timestamp_writes: None,
+    });
+    downsample_pass.set_pipeline(downsample_depth_first_pipeline);
+    downsample_pass.set_push_constants(
+        0,
+        &meshlet_view_resources.depth_pyramid_mip_count.to_le_bytes(),
+    );
+    downsample_pass.set_bind_group(0, &meshlet_view_bind_groups.downsample_depth, &[]);
+    downsample_pass.dispatch_workgroups(
+        meshlet_view_resources.view_size.x.div_ceil(64),
+        meshlet_view_resources.view_size.y.div_ceil(64),
+        1,
+    );
 
-        let mut downsample_pass = render_context.begin_tracked_render_pass(downsample_pass);
-        downsample_pass.set_render_pipeline(downsample_depth_pipeline);
-        downsample_pass.set_bind_group(0, &meshlet_view_bind_groups.downsample_depth[i], &[]);
-        downsample_pass.draw(0..3, 0..1);
+    if meshlet_view_resources.depth_pyramid_mip_count >= 7 {
+        downsample_pass.set_pipeline(downsample_depth_second_pipeline);
+        downsample_pass.dispatch_workgroups(
+            meshlet_view_resources.view_size.x.div_ceil(64),
+            meshlet_view_resources.view_size.y.div_ceil(64),
```

Comment:
Don't we want fewer workgroups for the second pass?

Reply:
Yes, fixed it 😅